src/endpointRoutes.js

```diff
@@ -38,6 +40,55 @@ const createEndpointRoute = router => {
   })
 }
 
+const readEndpointRoute = router => {
+  router.get('/endpoints/:endpointId', async (ctx, next) => {
+    const failureMsg = 'Retrieving of endpoint failed: '
+    const endpointId = ctx.params.endpointId
+
+    await readEndpoint(endpointId)
+      .then(endpoint => {
+        if (endpoint) {
+          ctx.status = 200
+          ctx.body = endpoint
+          logger.info(
+            `Endpoint "${endpoint.endpoint.pattern}" with pattern ${endpoint.endpoint.pattern} has been retrieved`
+          )
+        } else {
+          const error = `Endpoint with id ${endpointId} does not exist`
+          ctx.status = 404
+          ctx.body = {error: error}
+          logger.error(`${failureMsg}${error}`)
+        }
+        next()
+      })
+      .catch(error => {
+        ctx.statusCode = 400
+        handleServerError(ctx, failureMsg, error, logger)
+        next()
+      })
+  })
+}
+
+const readEndpointsRoute = router => {
+  router.get('/endpoints', async (ctx, next) => {
+    const failureMsg = 'Retrieving of endpoints failed: '
+
+    const queryParams = ctx.request.query
+
+    await readEndpoints(queryParams)
+      .then(endpoints => {
+        ctx.status = 200
+        ctx.body = endpoints
+        logger.info(`Endpoints have been successfully retrieved`)
```

Comment:
@bradsawadye I think this log could be a debug as there's no data change and would just be cluttering the logs. Also, if we add the `queryParam` it would give us a better idea of what data was retrieved.

```suggestion
        logger.debug(`Retrieved ${endpoints.length} Endpoints matching query param: ${JSON.stringify(queryParams)}`)
```